src/pages/auth/SignupPage.tsx

```diff
@@ -1,35 +1,26 @@
-import React, { useState } from "react";
 import {
   Box,
   Button,
   TextField,
   Typography,
   Container,
   Link as MuiLink,
-  Alert,
 } from "@mui/material";
 import { Link, useNavigate } from "react-router-dom";
 import { signup } from "../../utils/auth";
+import { useForm } from "react-hook-form";
+import { SignupCredentials } from "../../lib/types";
 
 export const SignupPage: React.FC = () => {
-  const [name, setName] = useState("");
-  const [email, setEmail] = useState("");
-  const [password, setPassword] = useState("");
-  const [error, setError] = useState("");
+  const { register, handleSubmit, formState: { errors } } = useForm<SignupCredentials>();
   const navigate = useNavigate();
 
-  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    setError("");
-    try {
-      const credentials = { name, email, password };
-      await signup(credentials);
-      navigate("/");
-    } catch (err) {
-      setError("Failed to create an account. Please try again.");
```

Comment:
Open an alert box `eg toast` on error or success

Reply:
I have added the toasters in the authentication utility. It displays proper toasts.